src/components/PopupWithForm.js

```diff
@@ -1,19 +1,33 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleFormSubmit, initialButtonText) {
     super({ popupSelector });
     this._popupForm = this._popupElement.querySelector(".modal__form");
     console.log(this._popupForm);
     this._handleFormSubmit = handleFormSubmit;
     console.log(this._closeButton);
+    this._submitButton = this._popupForm.querySelector(".modal__button");
+    this._initialButtonText = initialButtonText;
   }
 
   close() {
     this._popupForm.reset();
     super.close();
   }
 
+  renderLoading(isLoading) {
+    if (isLoading) {
+      if (this._initialButtonText === "Save") {
+        this._submitButton.textContent = "Saving...";
+      } else if (this._initialButtonText === "Create") {
+        this._submitButton.textContent = "Creating...";
+      }
+    } else {
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```